helm-chart/todoapp/templates/secret.yml

```diff
@@ -0,0 +1,20 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{.Chart.Name}}-secret
+  namespace: {{ .Values.todoapp.namespace }}
+type: Opaque
+data:
+{{- range $k, $v := .Values.todoapp.secret }}
+  {{- if eq $k "SECRET_KEY"}}
+  {{ $k }}: {{ $v }}
+  {{- else}}
+  {{ $k }}: {{ $v | quote }}
```

Comment:
Ensure that the values in `{{ .Values.todoapp.secret }}` are base64 encoded if necessary, especially for sensitive data. The `quote` function does not handle base64 encoding, so ensure that the values are already encoded before being templated here.